craft_providers/bases/buildd.py

```diff
@@ -1015,6 +1016,20 @@ def _update_apt_sources(
                 details=errors.details_from_called_process_error(error),
             ) from error
 
+        # update cloud.cfg to prevent the sources.list file from being reset
+        try:
+            _execute_run(
+                executor,
+                # 'aapt' is not a typo, the first 'a' is the sed command to append
+                # this is a shlex-compatible way to append to a file
+                ["sed", "-i", "$ aapt_preserve_sources_list: true", cloud_config],
```

Comment:
if cloud.cfg exists but is empty, this will exit with status 0 but not actually append to the file.

Reply:
Thanks, I added a check to make sure `cloud.cfg` exists and is not empty.